src/agent/communicator/include/https_socket.hpp

```diff
@@ -0,0 +1,100 @@
+#pragma once
+
+#include <ihttp_socket.hpp>
+
+#include <boost/asio.hpp>
+#include <boost/asio/experimental/awaitable_operators.hpp>
+#include <boost/asio/ssl.hpp>
+#include <boost/beast.hpp>
+#include <boost/system/error_code.hpp>
+
+#include <string>
+
+namespace http_client
+{
+    /// @brief Implementation of IHttpSocket for HTTPS requests
+    class HttpsSocket : public IHttpSocket
+    {
+    public:
+        /// @brief Constructor for HttpsSocket
+        /// @param ioContext The io context to use for the socket
+        HttpsSocket(const boost::asio::any_io_executor& ioContext)
+            : m_ctx(boost::asio::ssl::context::sslv23)
+            , m_ssl_socket(ioContext, m_ctx)
+        {
+            m_ctx.set_verify_mode(boost::asio::ssl::verify_peer);
+        }
+
+        /// @brief Connects the socket to the given endpoints
+        /// @param endpoints The endpoints to connect to
+        /// @param ec The error code, if any occurred
+        /// @param timeOut The timeout for the connection
+        void Connect(boost::asio::io_context& ioContext,
+                     const boost::asio::ip::tcp::resolver::results_type& endpoints,
+                     boost::system::error_code& ec,
+                     const std::chrono::seconds timeOut = std::chrono::seconds(TIMEOUT_DEFAULT)) override;
```

Comment:
Same note about the io context, the class has own, and is getting one more on each method. I understand it may not be the exact type needed to call `.run_for()`, we should still find a way to improve it.